apps/verticalhome/test/unit/sources/application_test.js

```diff
@@ -66,4 +67,35 @@ suite('app.js > ', function() {
     assert.ok(removeStub.calledTwice);
   });
 
+  test('blacklisted origin does not get added to grid', function() {
```

Comment:
nit, I'd like to see a test that makes sure that the blacklist works with more than one item in it.